File: Práctica3/Cola.py
class Cola:
    def __init__(self, n):
        self.n = n
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = 0

    def insert(self, e):
        if not self.isFull():
            self.arreglo[self.fin] = e
            self.fin += 1
        else:
            print("Cola llena")

    def remove(self):
        if not self.isEmpty():
            valor = self.arreglo[self.inicio]
            self.inicio += 1
            return valor
        else:
            print("Cola vacía")
            return None

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.inicio]
        else:
            print("Cola vacía")
            return None

    def isEmpty(self):
        return self.inicio == self.fin

    def isFull(self):
        return self.fin == self.n

    def size(self):
        return self.fin - self.inicio

File: Práctica3/Cola.java
package cola;

public class Cola {
    private long[] arreglo;
    private int inicio, fin, n;

    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = 0;
    }

    public void insert(long e) {
        if (!isFull()) {
            arreglo[fin++] = e;
        } else {
            System.out.println("Cola llena");
        }
    }

    public long remove() {
        if (!isEmpty()) {
            return arreglo[inicio++];
        } else {
            System.out.println("Cola vacía");
            return -1;
        }
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[inicio];
        } else {
            System.out.println("Cola vacía");
            return -1;
        }
    }

    public boolean isEmpty() {
        return inicio == fin;
    }

    public boolean isFull() {
        return fin == n;
    }

    public int size() {
        return fin - inicio;
    }
}